[WebXR] Remove PlatformXR::Instance::enumerateImmersiveXRDevices()

fujii opened the pull request:
#### 92620b8b587dda31f4650c2d4059f75788650d61
<pre>
[WebXR] Remove PlatformXR::Instance::enumerateImmersiveXRDevices()
<a href="https://bugs.webkit.org/show_bug.cgi?id=300977">https://bugs.webkit.org/show_bug.cgi?id=300977</a>

Reviewed by Sergio Villar Senin.

There were 3 implementations of enumerateImmersiveXRDevices() in the following
files:
- WebCore/platform/xr/cocoa/PlatformXRCocoa.mm
- WebCore/platform/xr/openxr/OpenXRInstance.cpp
- WebCore/platform/xr/openxr/PlatformXROpenXR.cpp

OpenXRInstance.cpp was used nowhere. The implementations of PlatformXRCocoa.mm
and PlatformXROpenXR.cpp just ruturned an empty list. Removed the files.

Renamed PlatformXR::Instance::DeviceList to PlatformXR::DeviceList.

* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::enumerateImmersiveXRDevices):
* Source/WebCore/platform/xr/PlatformXR.h:
* Source/WebCore/platform/xr/cocoa/PlatformXRCocoa.h: Removed.
* Source/WebCore/platform/xr/cocoa/PlatformXRCocoa.mm: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInstance.cpp: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h: Removed.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::enumerateImmersiveXRDevices):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/XR/PlatformXRSystemProxy.cpp:
(WebKit::PlatformXRSystemProxy::enumerateImmersiveXRDevices):
* Source/WebKit/WebProcess/XR/PlatformXRSystemProxy.h:

Canonical link: <a href="https://commits.webkit.org/301801@main">https://commits.webkit.org/301801@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e38749d0756a9c0e4a815023a7f43995baa6c2fe

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126810 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46451 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37426 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133758 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78423 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128681 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47077 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54987 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96497 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64527 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129758 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37670 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113433 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77016 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36546 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31610 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77152 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107493 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31915 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136338 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53483 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41171 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105004 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53979 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109792 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104708 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50213 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28557 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50946 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19879 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53412 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59214 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52668 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56002 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54416 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

svillar reviewed (approved):
Nice cleanup!. Those were all leftovers from the initial implementation of WebXR in which all the code was in WebCore. That was all moved to the UI process.

fujii committed:
[WebXR] Remove PlatformXR::Instance::enumerateImmersiveXRDevices()
https://bugs.webkit.org/show_bug.cgi?id=300977

Reviewed by Sergio Villar Senin.

There were 3 implementations of enumerateImmersiveXRDevices() in the following
files:
- WebCore/platform/xr/cocoa/PlatformXRCocoa.mm
- WebCore/platform/xr/openxr/OpenXRInstance.cpp
- WebCore/platform/xr/openxr/PlatformXROpenXR.cpp

OpenXRInstance.cpp was used nowhere. The implementations of PlatformXRCocoa.mm
and PlatformXROpenXR.cpp just ruturned an empty list. Removed the files.

Renamed PlatformXR::Instance::DeviceList to PlatformXR::DeviceList.

* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::enumerateImmersiveXRDevices):
* Source/WebCore/platform/xr/PlatformXR.h:
* Source/WebCore/platform/xr/cocoa/PlatformXRCocoa.h: Removed.
* Source/WebCore/platform/xr/cocoa/PlatformXRCocoa.mm: Removed.
* Source/WebCore/platform/xr/openxr/OpenXRInstance.cpp: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp: Removed.
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h: Removed.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::enumerateImmersiveXRDevices):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/XR/PlatformXRSystemProxy.cpp:
(WebKit::PlatformXRSystemProxy::enumerateImmersiveXRDevices):
* Source/WebKit/WebProcess/XR/PlatformXRSystemProxy.h:

Canonical link: https://commits.webkit.org/301801@main
